tests/Tests/ORM/Functional/Ticket/GH11982Test.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\DBAL\Platforms\SQLitePlatform;
+use Doctrine\Tests\OrmFunctionalTestCase;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\ORM\Mapping\ColumnIndex;
+use PHPUnit\Framework\Attributes\Group;
+
+class GH11982Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            GH11982ColumnIndex::class,
+        ]);
+    }
+
+    #[Group('GH-11982')]
+    public function testMetadata(): void
+    {
+        $metadata = $this->_em->getClassMetadata(GH11982ColumnIndex::class);
+        $indexes = $metadata->table['indexes'];
+
+        self::assertCount(3, $indexes);
+        self::assertEquals('indexTrue', $indexes[0]['fields'][0]);
+        self::assertArrayHasKey('class_idx', $indexes);
+        self::assertArrayHasKey('field_idx', $indexes);
+        self::assertEquals('test', $indexes['field_idx']['flags'][0]);
+        self::assertEquals('test', $indexes['field_idx']['options'][0]);
+    }
```

Comment:
This test is rather weak since it doesn't really confirm anything besides the table array has been modified.

If say `ORM\Index` has any changes in the future, this test will happily pass, even if that means no indices are added.

I just added it before I learnt about checking the actual schema, I left it in because it does check that the mapping is correct (flags, options,..), but I can understand if it's better to just add these checks to the next test and validate the actual SQL that is being generated instead.

Reply:
Even better than validating SQL, you could actually execute the SQL and instrospect the resulting schema.